lib/rules/selector-max-id/index.mjs

```diff
@@ -48,71 +45,57 @@ const rule = (primary, secondaryOptions) => {
 		}
 
 		/**
-		 * @param {import('postcss-selector-parser').Selector} resolvedSelectorNode
-		 * @param {import('postcss-selector-parser').Selector} selectorNode
-		 * @param {import('postcss').Rule} ruleNode
+		 * Return true if the node is inside an ignored context-functional pseudo-class
+		 *
+		 * @param {import('postcss-selector-parser').Node} node
+		 * @returns {boolean}
 		 */
-		function checkSelector(resolvedSelectorNode, selectorNode, ruleNode) {
-			const count = resolvedSelectorNode.reduce((total, childNode) => {
-				if (childNode.type === 'id') total += 1;
-
-				return total;
-			}, 0);
-
-			if (count > primary) {
-				const { index, endIndex, selector: selectorStr } = getStrippedSelectorSource(selectorNode);
-
-				report({
-					ruleName,
-					result,
-					node: ruleNode,
-					message: messages.expected,
-					messageArgs: [selectorStr, primary],
-					index,
-					endIndex,
-				});
+		function isInIgnoredContextFunctionalPseudoClass(node) {
+			/** @type {import('postcss-selector-parser').Node | import('postcss-selector-parser').Container | undefined} */
+			let current = node;
+
+			while (current) {
+				if (
+					current.type === 'pseudo' &&
```

Comment:
[suggestion] Minor. This `type === 'pseudo'` check seems unnecessary:

```suggestion
```

because `isContextFunctionalPseudoClass` already checks the same:

https://github.com/stylelint/stylelint/blob/b83a6375238b66e3b96280db5680a0fe96a74e78/lib/utils/isContextFunctionalPseudoClass.mjs#L14